File: NotesApp/NoteApp.swift
class NoteApp {

    private let textStorage = Storage<TextNoteModel>()
    private let reminderStorage = Storage<ReminderNoteModel>()
    private var notebook: Notebook!
    private var consoleUI: ConsoleUI!
    private var menu: Menu!
    

    func run() {
        setupDependencies()
        start()
    }
    

    private func setupDependencies() {

        notebook = Notebook(
            textNoteStorage: textStorage,
            reminderNoteStorage: reminderStorage
        )
        

        consoleUI = ConsoleUI()
        

        menu = Menu(console: consoleUI, notebook: notebook)
        

        setupTestData()
    }
    

    private func start() {
        print("Добро пожаловать в  Заметки!")
        menu.run()
    }
    

    private func setupTestData() {

        let textModel = TextNoteModel(
            id: "1",
            name: "Мои идеи",
            text: "Нужно реализовать крутое приложение"
        )
        
        let reminderModel = ReminderNoteModel(
            id: "2",
            text: "Сделать домашнее задание",
            isDone: false
        )
        

        let textNote = TextNote(data: textModel, storage: textStorage)
        let reminderNote = ReminderNote(data: reminderModel, storage: reminderStorage)
        
        notebook.add( textNote)
        notebook.add( reminderNote)
    }
}

File: BazaMobileHW/src/main/NotesApp/src/NoteApp.kt
class NoteApp {
    private val notebook = Notebook()
    private val consoleUI = ConsoleUI()
    private val menu = Menu(consoleUI, notebook)

    fun run() {
        setupTestData()
        start()
    }

    private fun start() {
        println("Добро пожаловать в Заметки!")
        menu.run()
    }

    private fun setupTestData() {
        val textNote = notebook.createTextNote("Мои идеи", "Нужно реализовать крутое приложение")
        val reminderNote = notebook.createReminderNote("Сделать домашнее задание")

        notebook.add(textNote)
        notebook.add(reminderNote)
    }
}